src/codegen/codegen_c_visitor.cpp

```diff
@@ -1779,13 +1785,16 @@ void CodegenCVisitor::visit_eigen_newton_solver_block(const ast::EigenNewtonSolv
     // try to use a different string for the matrices created by sympy in the form
     // X_<random_number>, J_<random_number>, Jm_<random_number> and F_<random_number>
     std::string X = find_var_unique_name("X");
+    std::string Xm = find_var_unique_name("Xm");
     std::string J = find_var_unique_name("J");
     std::string Jm = find_var_unique_name("Jm");
     std::string F = find_var_unique_name("F");
+    std::string Fm = find_var_unique_name("Fm");
 
     auto float_type = default_float_data_type();
     int N = node.get_n_state_vars()->get_value();
-    printer->add_line("Eigen::Matrix<{}, {}, 1> {};"_format(float_type, N, X));
+    printer->add_line("Eigen::Matrix<{}, {}, 1> {};"_format(float_type, N, Xm));
+    printer->add_line("{}* {} = {}.data();"_format(float_type, X, Xm));
```

Comment:
I was wondering what is the purpose of defining the variable `X`, since I don't see it being used

Reply:
The variables X, F & J are used in the generated code defined in :
```
print_statement_block(*node.get_initialize_block(), false, false);
print_statement_block(*node.get_setup_x_block(), false, false);
```
So all of them are used.

Reply:
Aha, I see now. Thanks for the clarification :+1: